src/Compilers/Core/Portable/CommandLine/CommandLineArguments.cs

```diff
@@ -73,72 +75,70 @@ public abstract class CommandLineArguments
         /// <summary>
         /// Compilation name or null if not specified.
         /// </summary>
-        public string CompilationName { get; internal set; }
+        public string? CompilationName { get; internal set; }
 
         /// <summary>
         /// Gets the emit options.
         /// </summary>
-        public EmitOptions EmitOptions { get; internal set; }
+        public EmitOptions EmitOptions { get; internal set; } = null!; // initialized by Parse
```

Comment:
❓ Why not make this a constructor parameter (required initialization)?

Reply:
Tried it after your suggestion, but decided against it as it breaks the regularity of the code in parser and also implies a change in VB code. Neither of these seem worth it.
The topic of field initializers will be re-discussed in LDM and at the very least we'll provide some guidance to the recommended pattern for dealing with this situation.

---
In reply to: [368190989](https://github.com/dotnet/roslyn/pull/41063#discussion_r368190989) [](ancestors = 368190989)